code_share/app_code_share/urls.py

```diff
@@ -2,7 +2,7 @@
 from .views import *
 
 urlpatterns = [
-    url(r'^$', home, name='home'),
-    url(r'^(?P<hash_id>\d+)/$', view_by_hash, name='view_by_hash'),
-    url(r'^(?P<file_name>[\w-]+)/$', view_by_file, name='view_by_file')
+    url(r'^$', home, name='app_home'),
+    url(r'^app/(?P<hash_id>\d+)/$', view_by_hash, name='view_by_hash'),
```

Comment:
I think it's we shouldn't have  _doman.com/app/<hash_id>_  but simply _domain.com/<hash_id>_ for simplicity. Whatcha think?